### PR TITLE
Clean up service package

### PR DESCRIPTION
**Pull request by peter-edge**
This is some more cleaning up of the service package. The primary point of this PR is to have public types before private types. This also removes some newlines and comments that seem unnecessary due to the code being self-documenting.

### CODE REVIEW

File: service/manager.go [hunk 1]
```diff
@@ -267,24 +283,8 @@ func (s *manager) addModule(provider ModuleProvider, options ...ModuleOption) er
...
-func (s *manager) Start() {
-	s.start()
-
-	// block until forced exit
-	s.WaitForShutdown(nil)
+	s.moduleWrappers = append(s.moduleWrappers, moduleWrapper)
+	return nil
 }
 
 func (s *manager) start() Control {
```
Comment (alsamylkin) [inline service/manager.go]: nit: could you remove the error on the following line?
I ran [grinder](https://github.com/rsc/grind) on our code and it was the only issue.

File: service/manager.go [hunk 2]
```diff
@@ -174,6 +141,55 @@ func (s *manager) OnCriticalError(err error) {
 	}
 }
 
+// StartAsync service is used as a non-blocking the call on service start. StartAsync will
+// return the call to the caller with a Control to listen on channels
+// and trigger manual shutdowns.
+func (s *manager) StartAsync() Control {
```
Comment (alsamylkin) [inline service/manager.go]: nit: should it be `m *manager`?:)

Comment (peter-edge) [inline service/manager.go]: Lol this was already there, but ya I will change it.

File: service/manager.go (hunk 1 above)
Comment (peter-edge) [inline service/manager.go]: Done

**Pull request merged**